### PR TITLE
refact, fix: Package managers

### PR DESCRIPTION
**Pull request by jeevithakannan2**
## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Documentation update
- [x] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
- Various package manager fixes.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/Developer-tools/githubdesktop-setup.sh [hunk 1]
```diff
@@ -9,20 +9,22 @@ installGithubDesktop() {
...
-                "$ESCALATION_TOOL" "$PACKAGER" install github-desktop
+                "$ESCALATION_TOOL" "$PACKAGER" update
+                "$ESCALATION_TOOL" "$PACKAGER" install -y github-desktop
                 ;;
             zypper)
                 "$ESCALATION_TOOL" rpm --import https://rpm.packages.shiftkey.dev/gpg.key
                 printf "%b\n" '[shiftkey-packages]\nname=GitHub Desktop\nbaseurl=https://rpm.packages.shiftkey.dev/rpm/\nenabled=1\ngpgcheck=1\nrepo_gpgcheck=1\ngpgkey=https://rpm.packages.shiftkey.dev/gpg.key\n' | "$ESCALATION_TOOL" tee /etc/zypp/repos.d/shiftkey-packages.repo > /dev/null
-                "$ESCALATION_TOOL" "$PACKAGER" ref && "$ESCALATION_TOOL" "$PACKAGER" install github-desktop
+                "$ESCALATION_TOOL" "$PACKAGER" refresh
+                "$ESCALATION_TOOL" "$PACKAGER" install -y github-desktop
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/githubdesktop-setup.sh:19]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install github-desktop
```

File: core/tabs/applications-setup/Developer-tools/vscodium-setup.sh [hunk 2]
```diff
@@ -6,21 +6,26 @@ installVsCodium() {
...
                 echo 'deb [ signed-by=/usr/share/keyrings/vscodium-archive-keyring.gpg ] https://download.vscodium.com/debs vscodium main' | "$ESCALATION_TOOL" tee /etc/apt/sources.list.d/vscodium.list
-                "$ESCALATION_TOOL" "$PACKAGER" install codium
+                "$ESCALATION_TOOL" "$PACKAGER" update
+                "$ESCALATION_TOOL" "$PACKAGER" install -y codium
                 ;;
             zypper)
                 "$ESCALATION_TOOL" rpmkeys --import https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo/-/raw/master/pub.gpg
                 printf "%b\n" "[gitlab.com_paulcarroty_vscodium_repo]\nname=gitlab.com_paulcarroty_vscodium_repo\nbaseurl=https://download.vscodium.com/rpms/\nenabled=1\ngpgcheck=1\nrepo_gpgcheck=1\ngpgkey=https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo/-/raw/master/pub.gpg\nmetadata_expire=1h" | "$ESCALATION_TOOL" tee -a /etc/zypp/repos.d/vscodium.repo
+                "$ESCALATION_TOOL" "$PACKAGER" refresh
+                "$ESCALATION_TOOL" "$PACKAGER" install -y codium
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/vscodium-setup.sh:19]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install codium
```

File: core/tabs/applications-setup/Developer-tools/neovim-setup.sh [hunk 3]
```diff
@@ -32,10 +32,7 @@ installNeovim() {
         apt-get|nala)
             "$ESCALATION_TOOL" "$PACKAGER" install -y neovim ripgrep fd-find python3-venv luarocks golang-go shellcheck git
             ;;
-        dnf)
-            "$ESCALATION_TOOL" "$PACKAGER" install -y neovim ripgrep fzf python3-virtualenv luarocks golang ShellCheck git
-            ;;
-        zypper)
+        dnf|zypper)
             "$ESCALATION_TOOL" "$PACKAGER" install -y neovim ripgrep fzf python3-virtualenv luarocks golang ShellCheck git
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/neovim-setup.sh:36]: we use `--non-interactive` for zypper

File: core/tabs/applications-setup/Developer-tools/sublime-setup.sh [hunk 4]
```diff
@@ -6,25 +6,25 @@ installSublime() {
...
-                    "$ESCALATION_TOOL" "$PACKAGER" install sublime-text
+                    "$ESCALATION_TOOL" "$PACKAGER" update
+                    "$ESCALATION_TOOL" "$PACKAGER" install -y sublime-text
                     ;;
                 zypper)
                     "$ESCALATION_TOOL" rpm -v --import https://download.sublimetext.com/sublimehq-rpm-pub.gpg
                     "$ESCALATION_TOOL" "$PACKAGER" addrepo -g -f https://download.sublimetext.com/rpm/dev/x86_64/sublime-text.repo
-                    "$ESCALATION_TOOL" "$PACKAGER" install sublime-text
+                    "$ESCALATION_TOOL" "$PACKAGER" refresh
+                    "$ESCALATION_TOOL" "$PACKAGER" install -y sublime-text
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/sublime-setup.sh:19]:
```suggestion
                    "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install sublime-text
```

File: core/tabs/applications-setup/Developer-tools/vscode-setup.sh [hunk 5]
```diff
@@ -11,21 +11,22 @@ installVsCode() {
...
-                "$ESCALATION_TOOL" "$PACKAGER" install code 
+                "$ESCALATION_TOOL" "$PACKAGER" update
+                "$ESCALATION_TOOL" "$PACKAGER" install -y apt-transport-https code
                 ;;
             zypper)
                 "$ESCALATION_TOOL" rpm --import https://packages.microsoft.com/keys/microsoft.asc
                 printf "%b\n" '[code]\nname=Visual Studio Code\nbaseurl=https://packages.microsoft.com/yumrepos/vscode\nenabled=1\ntype=rpm-md\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc' | "$ESCALATION_TOOL" tee /etc/zypp/repos.d/vscode.repo > /dev/null
-                "$ESCALATION_TOOL" "$PACKAGER" install code
+                "$ESCALATION_TOOL" "$PACKAGER" refresh
+                "$ESCALATION_TOOL" "$PACKAGER" install -y code
```
Comment (adamperkowski) [inline core/tabs/applications-setup/Developer-tools/vscode-setup.sh:21]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install code
```

File: core/tabs/applications-setup/android-debloat.sh [hunk 6]
```diff
@@ -9,13 +9,10 @@ install_adb() {
...
                 ;;
-            zypper)
-                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install android-tools
-                ;;
             pacman)
                 "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm android-tools
                 ;;
-            dnf)
+            dnf|zypper)
                 "$ESCALATION_TOOL" "$PACKAGER" install -y android-tools
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:16]: we use `--non-interactive` for zypper

File: core/tabs/applications-setup/browser-setup.sh [hunk 7]
```diff
@@ -39,12 +39,13 @@ install_thorium() {
...
                 "$ESCALATION_TOOL" rm -fv /etc/apt/sources.list.d/thorium.list
                 "$ESCALATION_TOOL" curl http://dl.thorium.rocks/debian/dists/stable/thorium.list -o /etc/apt/sources.list.d/thorium.list
+                "$ESCALATION_TOOL" "$PACKAGER" update
                 "$ESCALATION_TOOL" "$PACKAGER" install -y thorium-browser
                 ;;
             zypper|dnf)
                 url=$(curl -s https://api.github.com/repos/Alex313031/Thorium/releases/latest | grep -oP '(?<=browser_download_url": ")[^"]*\.rpm')
                     echo "$url" && curl -L "$url" -o thorium-latest.rpm
-                    "$ESCALATION_TOOL" rpm -i thorium-latest.rpm && rm thorium-latest.rpm
+                    "$ESCALATION_TOOL" "$PACKAGER" install -y thorium-latest.rpm && rm thorium-latest.rpm
```
Comment (adamperkowski) [inline core/tabs/applications-setup/browser-setup.sh:48]: we use `--non-interactive` for zypper

File: core/tabs/applications-setup/browser-setup.sh [hunk 8]
```diff
@@ -107,8 +109,8 @@ Signed-By: /usr/share/keyrings/librewolf.gpg" | "$ESCALATION_TOOL" tee /etc/apt/
 		zypper)
 			"$ESCALATION_TOOL" rpm --import https://rpm.librewolf.net/pubkey.gpg
 			"$ESCALATION_TOOL" zypper ar -ef https://rpm.librewolf.net librewolf
-			"$ESCALATION_TOOL" zypper ref
-			"$ESCALATION_TOOL" zypper in librewolf
+			"$ESCALATION_TOOL" zypper refresh
+			"$ESCALATION_TOOL" zypper install -y librewolf
```
Comment (adamperkowski) [inline core/tabs/applications-setup/browser-setup.sh:113]:
```suggestion
			"$ESCALATION_TOOL" zypper --non-interactive install librewolf
```

File: core/tabs/applications-setup/browser-setup.sh [hunk 9]
```diff
@@ -176,19 +180,11 @@ install_chromium() {
...
                 "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm chromium
                 ;;
-            dnf)
-                "$ESCALATION_TOOL" "$PACKAGER" install -y https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm
-                "$ESCALATION_TOOL" "$PACKAGER" install -y chromium
-                ;;
             *)
-                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
-                exit 1
+                "$ESCALATION_TOOL" "$PACKAGER" install -y chromium
```
Comment (adamperkowski) [inline core/tabs/applications-setup/browser-setup.sh:187]: add zypper here

File: core/tabs/applications-setup/communication-apps/telegram-setup.sh [hunk 10]
```diff
@@ -6,21 +6,11 @@ installTelegram() {
...
             pacman)
                 "$ESCALATION_TOOL" "$PACKAGER" -S --needed --noconfirm telegram-desktop 
                 ;;
-            dnf)
-                "$ESCALATION_TOOL" "$PACKAGER" install -y telegram-desktop 
-                ;;
             *)
-                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
-                exit 1
+                "$ESCALATION_TOOL" "$PACKAGER" install -y telegram-desktop
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/telegram-setup.sh:13]: add zypper here

File: core/tabs/applications-setup/communication-apps/thunderbird-setup.sh [hunk 11]
```diff
@@ -6,21 +6,11 @@ installThunderBird() {
...
             pacman)
                 "$ESCALATION_TOOL" "$PACKAGER" -S --needed --noconfirm thunderbird
                 ;;
-            dnf)
-                "$ESCALATION_TOOL" "$PACKAGER" install -y thunderbird 
-                ;;
             *)
-                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
-                exit 1
+                "$ESCALATION_TOOL" "$PACKAGER" install -y thunderbird
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/thunderbird-setup.sh:13]: add zypper

File: core/tabs/applications-setup/mybash-setup.sh [hunk 12]
```diff
@@ -11,18 +11,8 @@ installDepend() {
...
-        dnf)
-            "$ESCALATION_TOOL" "$PACKAGER" install -y bash bash-completion tar bat tree unzip fontconfig git
-            ;;
-        zypper)
-            "$ESCALATION_TOOL" "$PACKAGER" install -y bash bash-completion tar bat tree unzip fontconfig git
-            ;;
         *)
-            printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
-            exit 1
+            "$ESCALATION_TOOL" "$PACKAGER" install -y bash bash-completion tar bat tree unzip fontconfig git
```
Comment (adamperkowski) [inline core/tabs/applications-setup/mybash-setup.sh:15]: add zypper

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**